lib/Core/CorsProxy.ts

```diff
@@ -21,7 +21,7 @@ export default class CorsProxy {
    *  Domains that should be proxied for, as set by config files. Stored as an array of hosts - if a TLD is specified,
    * subdomains will also be proxied.
    */
-  proxyDomains: string[] | undefined = undefined;
+  proxyDomains: string[] = [];
```

Comment:
CorsProxySpec could be used almost without modifications by initializing proxyDomains to the empty array instead of undefined.

Allowing proxyDomains to be undefined means we have to insert a bunch of `if (isDefined(corsProxy.proxyDomain)) { }` around the content of the tests.